### PR TITLE
Enhancement Restore SQL

### PR DESCRIPTION
**Pull request by bijayprasadtiwari**
There are 2 fixes,
- The documentation for restore SQL while restoring to and alternate target.
- Added an option "TargetSecondaryDataFilesDirectoryList"  to restore in a secondary folder.

### CODE REVIEW

File: docs/cmdlets-reference/restore-cohesitymssqlobject.md [hunk 1]
```diff
@@ -252,6 +252,7 @@ Accept wildcard characters: False
 Specifies the secondary data filename pattern and corresponding directories of the DB.
 Secondary data files are optional and are user defined. The recommended file extension for secondary files is ".ndf".
 If this option is specified and the destination folders do not exist they will be automatically created.
+This field can be set only if restoring to a different target host.
```
Comment (ashish-cohesity) [inline docs/cmdlets-reference/restore-cohesitymssqlobject.md:255]: Can you put a todo here , we should support same host too unless that's api limitation

File: docs/cmdlets-reference/restore-cohesitymssqlobject.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Restore-CohesityMSSQLObject -HostSourceId <long> -JobId <long> -SourceId <long>
  [-CaptureTailLogs] [-JobRunId <long>] [-KeepOffline] [-NewDatabaseName <string>] [-NewInstanceName <string>]
  [-RestoreTimeSecs <long>] [-StartTime <long>] [-TargetDataFilesDirectory <string>]
  [-TargetHostCredential <PSCredential>] [-TargetHostId <long>] [-TargetHostParentId <long>]
- [-TargetLogFilesDirectory <string>] [<CommonParameters>]
+ [-TargetLogFilesDirectory <string>] [-TargetSecondaryDataFilesDirectoryList <List<FilenamePatternToDirectory>>] [<CommonParameters>]
```
Comment (ashish-cohesity) [inline docs/cmdlets-reference/restore-cohesitymssqlobject.md:13]: What is the api param called ? IMHO this is a long variable name.

File: src/Cohesity.Powershell/Cmdlets/Recovery/RestoreCohesityMSSQLObject.cs [hunk 3]
```diff
@@ -22,6 +22,21 @@ namespace Cohesity.Powershell.Cmdlets.Recovery
...
+    /// <example>
+    ///   <para>PS&gt;</para>
+    ///   <code>
+    /// $patternList = @()
+    /// $pattern = [Cohesity.Model.FilenamePatternToDirectory]::new()
+    /// $pattern.Directory = "C:\Secondary"
+    /// $pattern.FilenamePattern = ".ndf"
+    /// $patternList += $pattern
+    /// 
+    /// Restore-CohesityMSSQLObject -TaskName "restore-sql" -SourceId 698 -HostSourceId 675 -JobId 1359 -NewDatabaseName "restore-1" -NewInstanceName MSSQLSERVER -TargetHostId 972 -TargetDataFilesDirectory "C:\TEST Data" -TargetLogFilesDirectory "C:\TEST Log" -TargetSecondaryDataFilesDirectoryList $patternList
```
Comment (ashish-cohesity) [inline src/Cohesity.Powershell/Cmdlets/Recovery/RestoreCohesityMSSQLObject.cs:34]: whats the line length for powershell ? can you split this

File: docs/cmdlets-reference/restore-cohesitymssqlobject.md (hunk 1 above)
Comment (bijayprasadtiwari) [reply docs/cmdlets-reference/restore-cohesitymssqlobject.md:255]: This flag will not go with the same host.

File: docs/cmdlets-reference/restore-cohesitymssqlobject.md (hunk 2 above)
Comment (bijayprasadtiwari) [reply docs/cmdlets-reference/restore-cohesitymssqlobject.md:13]: The flag name is as is from the API. Its also self explanatory.

File: src/Cohesity.Powershell/Cmdlets/Recovery/RestoreCohesityMSSQLObject.cs (hunk 3 above)
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Cmdlets/Recovery/RestoreCohesityMSSQLObject.cs]: Done

**Pull request merged**